### ISSUE TITLE
[FEATURE REQUEST] Add option to generate skip link for screen reader users

### ISSUE THREAD

**Issue by kburk1997** [docsifyjs/docsify#494]
I am currently using docsify to generate organization documentation for an organization on my school's campus. I've noticed that when using a screen reader to test accessibility, it first tabs through the `aside` before going to the main content.

The current standard is to add a [skip link](https://webaim.org/techniques/skipnav/) to the main content.

I've tried adding a skip link manually with no luck, and it appears I have to dig into the docsify code in order to be able to add one.

That being said, an option to generate a skip link will really help with screen reader accessibility for documentation generated through this project.

**Comment by trusktr** [comment]
Sorry that no one replied to you since 2018! The project is being more actively maintained lately.

>  I have to dig into the docsify code in order to be able to add one.

It is currently a little difficult to modify the underlying source that generates the markup. Maybe we should change to a component-based system that encapsulates markup in each component.

But more back on topic, this is a good feature to add. @anikethsaha @jhildenbiddle Possible to add this in v4?

**Comment by jhildenbiddle** [comment]
Based on the [comments](https://github.com/docsifyjs/docsify/pull/495#issuecomment-390524354) in #495, the decision back in 2018 was to implement this feature as a third-party plugin. I'm guessing that this never happened given that a "skip link" plugin isn't listed on [awesome-docsify](https://github.com/docsifyjs/awesome-docsify).

Accessibility is something we should absolutely consider moving forward and do a better job with in v5. Regarding this particular issue, it's important to realize that the linked document describing best practices for "skip links" was last updated on October 23, 2013. Time have obviously changed since then, and so have accessibility best practices. I think we should better understand these best practices before we agree to add skip links by default. If there is something quick and low-risk that we can do for v4, great. If the work is more involved, let's create a PR and target it for v5.

@kburk1997 -- Are you still using docsify for your organization? If so, do you still have a need for "skip link" functionality or have you worked around this issue?

**Comment by kburk1997** [comment]
Apologies for the late reply - I have graduated the university I was attending when I first opened the issue. I have since found another use case for docsify - is there any update on adding a skip link?

**Comment by jhildenbiddle** [comment]
Thanks for creating the issue and for your patience white waiting for it to be addressed, @kburk1997.

I've created a draft PR here: #2253. In that PR you will find a preview link which will allow you to review the changes:

![CleanShot 2023-10-17 at 08 59 27@2x](https://github.com/docsifyjs/docsify/assets/442527/7e4bf479-b02c-46c3-8e98-a99ffefa1ae5)

Before I mark this PR as ready for review by the maintainers, I would love to get your thoughts and/or feedback.

Thanks!
